modules/core/src/v2/coins/hbar.ts

```diff
@@ -0,0 +1,221 @@
+  SignedTransaction,
+  SignTransactionOptions,
+  VerifyAddressOptions,
+  VerifyTransactionOptions,
+  TransactionFee,
+  TransactionRecipient as Recipient,
+  TransactionPrebuild as BaseTransactionPrebuild,
+  TransactionExplanation,
+} from '../baseCoin';
+
+import { BitGo } from '../../bitgo';
+import { NodeCallback } from '../types';
+import { MethodNotImplementedError } from '../../errors';
+
+export interface HbarSignTransactionOptions extends SignTransactionOptions {
+  txPrebuild: TransactionPrebuild;
+  prv: string;
+}
+
+export interface TxInfo {
+  recipients: Recipient[];
+  from: string;
+  txid: string;
+}
+
+export interface TransactionPrebuild extends BaseTransactionPrebuild {
+  txHex: string;
+  txInfo: TxInfo;
+  feeInfo: TransactionFee;
+  source: string;
+}
+
+export interface ExplainTransactionOptions {
+  txHex?: string;
+  halfSigned?: {
+    txHex: string;
+  };
+  feeInfo: TransactionFee;
+}
+
+export class Hbar extends BaseCoin {
+  protected readonly _staticsCoin: Readonly<StaticsBaseCoin>;
+
+  constructor(bitgo: BitGo, staticsCoin?: Readonly<StaticsBaseCoin>) {
+    super(bitgo);
+
+    if (!staticsCoin) {
+      throw new Error('missing required constructor parameter staticsCoin');
+    }
+
+    this._staticsCoin = staticsCoin;
+  }
+
+  getChain() {
+    return this._staticsCoin.name;
+  }
+
+  getFamily(): CoinFamily {
+    return this._staticsCoin.family;
+  }
+
+  getFullName() {
+    return this._staticsCoin.fullName;
+  }
+
+  getBaseFactor() {
+    return Math.pow(10, this._staticsCoin.decimalPlaces);
+  }
+
+  static createInstance(bitgo: BitGo, staticsCoin?: Readonly<StaticsBaseCoin>): BaseCoin {
+    return new Hbar(bitgo, staticsCoin);
+  }
+
+  /**
+   * Flag for sending value of 0
+   * @returns {boolean} True if okay to send 0 value, false otherwise
+   */
+  valuelessTransferAllowed(): boolean {
+    return false;
+  }
+
+  /**
+   * Checks if this is a valid base58 or hex address
+   * @param address
+   */
+  isValidAddress(address: string): boolean {
+    if (!address) {
+      return false;
+    }
+    return bitgoAccountLib.Hbar.Utils.isValidAddress(address);
+  }
+
+  /**
+   * Generate Hedera Hashgraph key pair
+   *
+   * @param seed
+   * @returns {Object} object with generated pub, prv
+   */
+  generateKeyPair(seed?: Buffer): KeyPair {
+    const keyPair = seed ? new bitgoAccountLib.Hbar.KeyPair({ seed }) : new bitgoAccountLib.Hbar.KeyPair();
+    const keys = keyPair.getKeys();
+
+    if (!keys.prv) {
+      throw new Error('Keypair generation failed to generate a prv');
+    }
+
+    return {
+      pub: Buffer.from(keys.pub).toString('hex'),
+      prv: Buffer.from(keys.prv).toString('hex'),
+    };
+  }
+
+  parseTransaction(
+    params: ParseTransactionOptions,
+    callback?: NodeCallback<ParsedTransaction>
+  ): Bluebird<ParsedTransaction> {
+    return Bluebird.resolve({}).asCallback(callback);
+  }
+
+  verifyAddress(params: VerifyAddressOptions): boolean {
+    return true;
```

Comment:
Shouldn't this use `bitgoAccountLib.Hbar.Utils.isValidAddress(address)`? Or how do we use this method?

Reply:
the implementation is basically required - it's used for verifying an address is on a wallet - I changed it so that it throws an error now instead of blindly verifying

Reply:
eh you know any memoId is basically valid on a wallet